lib/app.js

```diff
@@ -1,24 +1,34 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types'
 import shallowEquals from './shallow-equals'
-import { warn } from './utils'
+import { warn, loadGetInitialProps } from './utils'
 import { makePublicRouterInstance } from './router'
 
 export default class App extends Component {
   state = {
     hasError: null
   }
 
+  static displayName = 'App'
+
+  static async getInitialProps ({ Component, router, ctx }) {
```

Comment:
If user didn't override `getInitialProps` in his/her sub class, is this going to execute?